django/conf/locale/__init__.py

```diff
@@ -587,12 +587,42 @@
         'name': 'Simplified Chinese',
         'name_local': '简体中文',
     },
+    'zh-Hans-CN': {
+        'fallback': ['zh-hans'],
+    },
+    'zh-Hans-HK': {
+        'fallback': ['zh-hans'],
+    },
+    'zh-Hans-MO': {
+        'fallback': ['zh-hans'],
+    },
+    'zh-Hans-SG': {
+        'fallback': ['zh-hans'],
+    },
+    'zh-Hans-TW': {
+        'fallback': ['zh-hans'],
+    },
     'zh-hant': {
         'bidi': False,
         'code': 'zh-hant',
         'name': 'Traditional Chinese',
         'name_local': '繁體中文',
     },
+    'zh-Hant-CN': {
+        'fallback': ['zh-hant'],
+    },
+    'zh-Hant-HK': {
+        'fallback': ['zh-hant'],
+    },
+    'zh-Hant-MO': {
+        'fallback': ['zh-hant'],
+    },
+    'zh-Hant-SG': {
+        'fallback': ['zh-hant'],
+    },
+    'zh-Hant-TW': {
+        'fallback': ['zh-hant'],
+    },
```

Comment:
Please add new fallabacks to the `tests.i18n.tests.MiscTests.test_special_fallback_language()` (use `self.subTest()`).

Reply:
Is this ok? Also, should it contain tests with lower case language codes?
```diff
diff --git a/tests/i18n/tests.py b/tests/i18n/tests.py
index 8248122a97..69c7cd968f 100644
--- a/tests/i18n/tests.py
+++ b/tests/i18n/tests.py
@@ -1419,10 +1419,25 @@ class MiscTests(SimpleTestCase):
         Some languages may have special fallbacks that don't follow the simple
         'fr-ca' -> 'fr' logic (notably Chinese codes).
         """
-        r = self.rf.get('/')
-        r.COOKIES = {}
-        r.META = {'HTTP_ACCEPT_LANGUAGE': 'zh-my,en'}
-        self.assertEqual(get_language_from_request(r), 'zh-hans')
+        tests = [
+            ('zh-my', 'zh-hans'),
+            ('zh-Hans-CN', 'zh-hans'),
+            ('zh-Hans-HK', 'zh-hans'),
+            ('zh-Hans-MO', 'zh-hans'),
+            ('zh-Hans-SG', 'zh-hans'),
+            ('zh-Hans-TW', 'zh-hans'),
+            ('zh-Hant-CN', 'zh-hant'),
+            ('zh-Hant-HK', 'zh-hant'),
+            ('zh-Hant-MO', 'zh-hant'),
+            ('zh-Hant-SG', 'zh-hant'),
+            ('zh-Hant-TW', 'zh-hant'),
+        ]
+        for value, expected in tests:
+            with self.subTest(value=value):
+                r = self.rf.get('/')
+                r.COOKIES = {}
+                r.META = {'HTTP_ACCEPT_LANGUAGE': f'{value},en'}
+                self.assertEqual(get_language_from_request(r), expected)
 
     def test_parse_language_cookie(self):
         """
```